services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -71,19 +81,25 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
   override def getFiles(project: Project, path: Path)(implicit ec: ExecutionContext): AsyncResult[List[String]] = ???
 
   override def getProjectVersions(project: Project)(implicit ec: ExecutionContext): AsyncResult[Seq[Version]] = {
+
     val versionsListUrl: String = s"${config.url}projects/${project.repository.get.value}/repository/tags"
     httpClient
-      .get(url = versionsListUrl, headers = config.token)
+      .get[Seq[Version]](url = versionsListUrl, headers = config.token)
       .map(
         resp =>
-          if (resp.status != HttpStatusCodes.OK)
-            Left(VersioningException(s"Could not take versions. Response status: ${resp.status}"))
-          else {
-            val parsedVersions: JsResult[Seq[Version]] = Json.parse(resp.body).validate[Seq[Version]]
-            parsedVersions match {
-              case JsSuccess(value, _) => Right(value)
-              case JsError(errors)     => Left(VersioningException(s"Could not parse GitLab response. (errors: $errors)"))
-            }
+          resp.status match {
+            case HttpStatusCodes.OK =>
```

Comment:
I think there is no need to pattern match on response status. Pattern matching on the response body should be enough.
However, if you think that you want two pattern matches on both of these things, do not use nested pattern matching. You did it right on lines 40-42.

Reply:
Corrected. Very useful correction, thanks! I will try to use this later!